src/Config.Net/TypeParsers/DictionaryParser.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace Config.Net.TypeParsers
+{
+   class DictionaryParser : ITypeParser
+   {
+      public IEnumerable<Type> SupportedTypes => new[] { typeof(Dictionary<string, string>), typeof(Dictionary<string, object>) };
+
+      public bool TryParse(string value, Type t, out object result)
+      {
+         if (value == null)
+         {
+            result = null;
+            return false;
+         }
+
+         result = JsonConvert.DeserializeObject(value, t);
```

Comment:
We might need some validation here. If you are supporting only `Dictionary<string, string>` and `Dictionary<string, object>` not sure what happens if `value` contains a string which deserializes into something else. At the moment library's behaviour is to return null and false for types it can't parse. Maybe worth adding a test for this as well.